src/hostapi/asio/pa_asio.cpp

```diff
@@ -1871,24 +1874,50 @@ static PaError ValidateAsioSpecificStreamInfo(
 {
     if( streamInfo )
     {
-        if( streamInfo->size != sizeof( PaAsioStreamInfo )
-                || streamInfo->version != 1 )
+        switch( streamInfo->version )
         {
+        case 0:
+            /* NOTE: Struct version 0 is invalid, possibly indicative of uninitialized data. */
             return paIncompatibleHostApiSpecificStreamInfo;
+
+        case 1:
+            /* NOTE: V1 structure's size is smaller by one pointer. */
+            if( streamInfo->size < sizeof( PaAsioStreamInfo ) - sizeof(PaAsio_MessageCallback*) )
```

Comment:
Structly speaking, `<` is incorrect. If when defining the v2 struct, the v1 struct has to be padded to append an aligned `PaAsio_MessageCallback*` then the v1 struct will be less than `sizeof( PaAsioStreamInfo ) - sizeof(PaAsio_MessageCallback*)`. In practice this is probably not an issue since `alignof(v1struct)` is probably equal to `alignof(PaAsio_MessageCallback*)`.

Reply:
Alternatives would be to declare a V1 version of the structure, formulate a more elaborate size test based on normative compiler behavior, or not test at all.  The previous field is a pointer too.  So I'm inclined to leave this imperfect heuristic as it is.

Reply:
I think we all agree that it is imperfect but we'll live with it for now.